src/launch/main.c

```diff
@@ -464,7 +503,7 @@ static int manager_start_transient_unit(Manager *manager, Service *service) {
         if (r < 0)
                 return error_origin(r);
 
-        r = sd_bus_send(manager->bus_regular, method_call, NULL);
+        r = sd_bus_call_async(manager->bus_regular, &service->slot, method_call, manager_start_unit_handler, service, -1);
```

Comment:
Ehhh, in both cases, you sure this is safe? I mean, what guarantees you that `service->slot` is NULL and can be re-used?

Imagine this: Name is activated. We get the signal and send the request to systemd. In The meantime, the name is dropped and activated again. We get the signal again and will reuse `service->slot` before it was cleared by the reply handler.

I think the right solution is to do `service->slot = sd_bus_slot_unref(service->slot);` right at the top of `manager_start_unit()`. So we discard the old request and send a new-one. We cannot just skip it, since someone might rely on a barrier and changed systemd units etc. So we must send a new request to guarantee ordering.

Reply:
Broker will not send more than one request at a time, but you are right, the service can be activated, grab its name, release it again, and someone can activate the name again, before we get the answer to StartUnit() from systemd, and this would leak the slot. I'll do clear it as you suggested.